docbrown/db/src/graph.rs

```diff
@@ -63,6 +64,19 @@ impl Graph {
         WindowedGraph::new(self.clone(), t_start, t_end)
     }
 
+    pub fn through_perspectives(&self, mut perspectives: PerspectiveSet) -> GraphWindowSet {
+        let timeline = match (self.earliest_time(), self.latest_time()) {
+            (Some(start), Some(end)) => start..end,
+            _ => 0..0,
+        };
+        let iter = perspectives.build_iter(timeline);
```

Comment:
Should this just return an empty iterator if there is no data in the graph.

Can we test an empty graph?

Reply:
fixed